### PR TITLE
Made eigenvalue system support scalar kernels

### LINKED ISSUE #13549: Eigenvalue system suports scalar kernels
## Reason

There are some applications that need to construct some eigenvalue system for scalar kernels.

## Design

Eigenvalue system should be able to compute the eigenvalues for the system from scalar kernels.
## Impact

Add something new.

### PR DESCRIPTION
**Pull request by fdkong**
Closes #13549

### CODE REVIEW

File: framework/src/systems/NonlinearEigenSystem.C [hunk 1]
```diff
@@ -140,10 +140,12 @@ void
 NonlinearEigenSystem::initialSetup()
 {
   NonlinearSystemBase::initialSetup();
-
+  // Regular kernels
   addEigenTagToMooseObjects(_kernels);
-
+  // Nodal BCs (we do not care about IBCs)
```
Comment (lindsayad) [inline framework/src/systems/NonlinearEigenSystem.C]: Why don't we care?

**Comment by loganharbour** [comment]
@fdkong, zero matrix:
```
%Mat Object: () 1 MPI processes
%  type: seqaij
% Size = 2 2
% Nonzeros = 0
zzz = zeros(1,3);
zzz = [
2 2  0.0000000000000000e+00
];
```

**Comment by fdkong** [comment]
> @fdkong, zero matrix:
> 
> ```
> %Mat Object: () 1 MPI processes
> %  type: seqaij
> % Size = 2 2
> % Nonzeros = 0
> zzz = zeros(1,3);
> zzz = [
> 2 2  0.0000000000000000e+00
> ];
> ```

It looks like working on my desktop. It is also working in test boxes.

**Comment by loganharbour** [comment]
Didn't work with a clobberall and libmesh rebuild. Works now after deleting and re-cloning.

File: framework/src/systems/NonlinearEigenSystem.C (hunk 1 above)
Comment (fdkong) [reply framework/src/systems/NonlinearEigenSystem.C]: By the definition of eigenvalue problems, we require the integral BCs equal zero. And then the integral BCs won't do anything at all in the eigenvalue problem.

Comment (YaqiWang) [reply framework/src/systems/NonlinearEigenSystem.C]: @fdkong This is not true. I guess you have homogeneous Neumann BC in mind, which is indeed naturally imposed for diffusion problem. Eigenvalue problems should allow us to use something like homogeneous Robin BC. This issue was brought up by @loganharbour yesterday. I think we should have a virtual function `bool BoundaryCondition::isHomogeneous() const`, that can be used for integrity check in eigen problems.

Comment (fdkong) [reply framework/src/systems/NonlinearEigenSystem.C]: I do not think I understand Robin BCs for eigenvalue problems. 

Could you make an example?

Comment (YaqiWang) [reply framework/src/systems/NonlinearEigenSystem.C]: https://github.com/idaholab/moose/blob/next/framework/src/bcs/VacuumBC.C is actually the Robin BC.

Comment (fdkong) [reply framework/src/systems/NonlinearEigenSystem.C]: Got it.

**Review by lindsayad — APPROVED** [review]
This PR satisfies its title. It sounds like in the future we may add support for IntegratedBCs

**Comment by YaqiWang** [comment]
Yes, someone needs to create an issue, @loganharbour do you want to?

**Pull request merged**